benchmarks/lasso/solvers/baseline.py

```diff
@@ -5,33 +5,34 @@
 
 
 class Solver(BaseSolver):
-    name = 'Baseline'
+    name = 'Baseline'  # proximal gradient, eventually accelerated
 
+    # any parameter defined here is accessible as a class attribute
     parameters = {'use_acceleration': [False, True]}
 
     def set_objective(self, X, y, lmbd):
         self.X, self.y, self.lmbd = X, y, lmbd
 
-        self.L = np.linalg.norm(self.X.dot(self.X.T), ord=2)
+        self.L = np.linalg.norm(self.X, ord=2) ** 2
 
     def run(self, n_iter):
         n_features = self.X.shape[1]
-        z_hat = np.zeros(n_features)
+        w = np.zeros(n_features)
         t_new = 1
         for _ in range(n_iter):
             if self.use_acceleration:
                 t_old = t_new
                 t_new = (1 + np.sqrt(1 + 4 * t_old ** 2)) / 2
-                z_old = z_hat.copy()
-            grad = self.X.T.dot(self.X.dot(z_hat) - self.y)
-            z_hat -= 1 / self.L * grad
-            z_hat = self.st(z_hat, 1 / self.L * self.lmbd)
+                w_old = w.copy()
+            grad = self.X.T.dot(self.X.dot(w) - self.y)
+            w -= 1 / self.L * grad
+            w = self.st(w, 1 / self.L * self.lmbd)
             if self.use_acceleration:
-                z_hat = z_hat + (t_old - 1.) / t_new * (z_hat - z_old)
-        self.z_hat = z_hat
+                w = w + (t_old - 1.) / t_new * (w - w_old)
```

Comment:
you should use inplace operations as much as possible.

Reply:
done